### PR TITLE
Латышев Алексей Сергеевич, группа 2539

### PR DESCRIPTION
**Pull request by alex-700**
Все части.

### CODE REVIEW

**Comment by thevery** [comment]
__MACOSX folder should not be commited

**Comment by thevery** [comment]
weather for "Dachnoye" (auto-detected) doesn't work: http://api.openweathermap.org/data/2.5/forecast/daily?q=Dachnoye&mode=json&units=metric&cnt=7&APPID=d690342f9a15003127f67bc6aff8418c

File: app/src/main/java/com/alex700/AWeather/CityManager.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                    Log.d("init", "start");
+                    for (String s = br.readLine(); s != null; s = br.readLine()) {
+                        strings.add(s);
+                    }
+                    Log.d("init", "stop");
+                    return strings;
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    return null;
+                }
```
Comment (thevery) [inline app/src/main/java/com/alex700/AWeather/CityManager.java:46]: close in finally?

File: app/src/main/java/com/alex700/AWeather/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.Objects;
+
+
+public class MainActivity extends Activity
+        implements NavigationDrawerFragment.NavigationDrawerCallbacks,
+        WeatherFragment.OnFragmentInteractionListener
+{
```
Comment (thevery) [inline app/src/main/java/com/alex700/AWeather/MainActivity.java:42]: why on separate line?

File: app/src/main/java/com/alex700/AWeather/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+    public static final String SP_NAME = "WEATHER";
+    /**
+     * Fragment managing the behaviors, interactions and presentation of the navigation drawer.
+     */
+    private NavigationDrawerFragment mNavigationDrawerFragment;
```
Comment (thevery) [inline app/src/main/java/com/alex700/AWeather/MainActivity.java:48]: inconsistent naming

File: app/src/main/java/com/alex700/AWeather/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+        // Set up the drawer.
+        mNavigationDrawerFragment.setUp(
+                R.id.navigation_drawer,
+                (DrawerLayout) findViewById(R.id.drawer_layout));
+        Location lastKnownLocation = ((LocationManager) getSystemService(Context.LOCATION_SERVICE))
+                .getLastKnownLocation(LocationManager.PASSIVE_PROVIDER);
+        if (lastKnownLocation != null) {
+            Log.d("LOCATION", "start");
+            Toast.makeText(this, "start", Toast.LENGTH_SHORT).show();
```
Comment (thevery) [inline app/src/main/java/com/alex700/AWeather/MainActivity.java:73]: `strings.xml`!

File: app/src/main/java/com/alex700/AWeather/NavigationDrawerFragment.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+                    ((ArrayAdapter) ((ListView) getActivity().findViewById(R.id.city_list)).getAdapter()).notifyDataSetChanged();
+                    selectItem(0);
+                } else {
+                    Toast.makeText(getActivity(), "Don't delete last town !", Toast.LENGTH_SHORT).show();
+                }
+                return true;
+            }
+        });
+        ((Button) mDrawerListView.findViewById(R.id.add_city_button)).setOnClickListener(this);
+        getLoaderManager().restartLoader(2222, null, this);
```
Comment (thevery) [inline app/src/main/java/com/alex700/AWeather/NavigationDrawerFragment.java:137]: const!

**Comment by alex-700** [comment]
It's a pity. I checked it on Kronverkskiy, Aptekarskiy and Juravli (Samara region). It worked fine.

**Comment by thevery** [comment]
- add city EditText is not single line
- where is no need to "=" in UI
- too much toasts

**Comment by thevery** [comment]
ok
24-6=18

**Pull request closed without merging**